### DISCUSSION TITLE
How to show a progress bar while route changes using React concurrent mode? 

### DISCUSSION THREAD

**Discussion by videni** [vercel/next.js#10815] [Help]
I am using React concurrent mode, when route changes, I'd like to stay at current page for a while and also show a progress bar at the top of the web page. any way to use `Router.push` with `useTransition`
```
const useTransition = React.useTransition;
```

**Comment by willemliu** [comment]
Have you tried Router events?
https://nextjs.org/docs/api-reference/next/router#routerevents
